include/tinycompress/tinycompress.h

```diff
@@ -307,6 +307,38 @@ int is_compress_ready(struct compress *compress);
 /* Returns a human readable reason for the last error */
 const char *compress_get_error(struct compress *compress);
 
+/*
+ * compress_get_caps: gets the list of supported codecs
+ *
+ * returns number of codecs on success, negative on error
+ *
+ * @card: sound card number
+ * @device: device number
+ * @flags: stream flags
+ * @codecs: Pointer to an array which will hold the list of codecs
+ * @size: Size of the codecs array. Should be enough to hold the list of
+ * supported codecs.
+ */
+int compress_get_supported_codecs(unsigned int card, unsigned int device,
+		unsigned int flags, int *codecs, int size);
+
+/*
+ * compress_get_caps: gets the list of supported codecs
+ *
+ * returns number of codecs on success, negative on error
+ *
+ * format of name is :
+ *    hw:<card>,<device> for real hw compress node
+ *    <plugin_libname>:<custom string> for virtual compress node
+ *
+ * @name: name of the compress node
+ * @flags: stream flags
+ * @codecs: Pointer to an array which will hold the list of codecs
+ * @size: Size of the codecs array. Should be enough to hold the list of
+ * supported codecs.
```

Comment:
and how would you know what the size is? There's no API. You would need something that returns a count first, so that the memory allocation is done appropriately.
Alternatively this should be the maximum size.

Reply:
Have added an API to return the actual count of audio codecs.